src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,72 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../services/api';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
```

Comment:
Ensure that `state.currentTodo` is correctly defined in your Redux store. If `currentTodo` is not part of the state, this line will cause an error. Verify the structure of your Redux state to ensure this selector works as intended.